<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="vanBuggenum" />


<title>Identifying cell-states</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manuscript docs ASCs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="QC.html">QC</a>
</li>
<li>
  <a href="cellstate.html">ASC characterization</a>
</li>
<li>
  <a href="hallmarks.html">Hallmarks Ig-classes</a>
</li>
<li>
  <a href="license.html">Citation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Identifying cell-states</h1>
<h4 class="author">vanBuggenum</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Multimodal-Plasmacell_manuscript/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211005code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211005)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211005code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211005)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumMultimodalPlasmacellmanuscripttree95e922e0e923cc8661dd886c583394eeed3cd82atargetblank95e922ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript/tree/95e922e0e923cc8661dd886c583394eeed3cd82a" target="_blank">95e922e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomvanbuggenumMultimodalPlasmacellmanuscripttree95e922e0e923cc8661dd886c583394eeed3cd82atargetblank95e922ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript/tree/95e922e0e923cc8661dd886c583394eeed3cd82a" target="_blank">95e922e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/cellstate_sidetest.Rmd
    Ignored:    analysis/hallmarks2.Rmd
    Ignored:    analysis/supplements.Rmd
    Ignored:    data/Seq2Science/
    Ignored:    data/azimuth_PBMCs/
    Ignored:    data/azimuth_bonemarrow/
    Ignored:    data/citeseqcount_htseqcount.zip
    Ignored:    data/genelist.plots.diffmarkers.txt
    Ignored:    data/genelist.plots.diffmarkers2.txt
    Ignored:    data/raw/
    Ignored:    data/supplementary/
    Ignored:    output/MOFA_analysis_Donorgroup.hdf5
    Ignored:    output/MOFA_analysis_Donorgroup.rds
    Ignored:    output/MOFA_analysis_Donorgroup_clustered.rds
    Ignored:    output/MOFA_analysis_Donorgroup_noIg.hdf5
    Ignored:    output/MOFA_analysis_Donorgroup_noIg2.hdf5
    Ignored:    output/extra plots.docx
    Ignored:    output/paper_figures/
    Ignored:    output/seu.fix_norm.rds
    Ignored:    output/seu.fix_norm_cellstate.rds
    Ignored:    output/seu.fix_norm_plasmacells.rds
    Ignored:    output/seu.live_norm.rds
    Ignored:    output/seu.live_norm_cellstate.rds
    Ignored:    output/seu.live_norm_plasmacells.rds
    Ignored:    output/seu.live_norm_plasmacells_RNA.rds
    Ignored:    output/top-PROT-loadings_IgA.tsv
    Ignored:    output/top-PROT-loadings_IgG.tsv
    Ignored:    output/top-PROT-loadings_IgM.tsv
    Ignored:    output/top-gene-loadings_IgA.tsv
    Ignored:    output/top-gene-loadings_IgG.tsv
    Ignored:    output/top-gene-loadings_IgM.csv
    Ignored:    output/top-gene-loadings_IgM.tsv

Unstaged changes:
    Modified:   .gitignore
    Modified:   CITATION.bib

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/cellstate.Rmd</code>) and HTML
(<code>docs/cellstate.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript/blob/95e922e0e923cc8661dd886c583394eeed3cd82a/analysis/cellstate.Rmd" target="_blank">95e922e</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2023-01-17
</td>
<td>
final docs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/Multimodal-Plasmacell_manuscript/b564113a0388c0d0621b868e2aab1ba6fed356bb/docs/cellstate.html" target="_blank">b564113</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2022-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript/blob/67dc2d2eb6d149f32bcec157da2e093fc9c0e923/analysis/cellstate.Rmd" target="_blank">67dc2d2</a>
</td>
<td>
Jessie van Buggenum
</td>
<td>
2022-12-16
</td>
<td>
Fig 1 and 2 and supplementary figures documentation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/vanbuggenum/Multimodal-Plasmacell_manuscript/b6ab01be103ff76870a4281e316ddf23b5ed5f91/docs/cellstate.html" target="_blank">b6ab01b</a>
</td>
<td>
jessievb
</td>
<td>
2021-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/vanbuggenum/Multimodal-Plasmacell_manuscript/blob/791178a412ef69330afc63879f4e1114581c14c3/analysis/cellstate.Rmd" target="_blank">791178a</a>
</td>
<td>
jessievb
</td>
<td>
2021-12-06
</td>
<td>
Plasmacell and cell-cycle gating analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>seu.fix &lt;- readRDS( file = &quot;output/seu.fix_norm.rds&quot;)
seu.live &lt;- readRDS(file = &quot;output/seu.live_norm.rds&quot;)</code></pre>
<div id="summary" class="section level2">
<h2>Summary</h2>
<ul>
<li>Cell-Cycle scoring<br />
</li>
<li>Plasma cell markers (CD27high &amp; IgDlow)<br />
</li>
<li>Ig-subtype classification</li>
</ul>
</div>
<div id="cell-cycle-state" class="section level2">
<h2>Cell-cycle state</h2>
<p>Cell-cycle scoring was performed scoring algorithm <a
href="https://carmonalab.github.io/UCell/UCell_Seurat_vignette.html">UCell</a>.
(For citation see (<a
href="https://www.biorxiv.org/content/10.1101/2021.04.13.439670v1">this</a>
preprint)</p>
<pre class="r"><code>s.genes &lt;- cc.genes.updated.2019$s.genes
g2m.genes &lt;- cc.genes.updated.2019$g2m.genes

###          --- Live cells ------

# Cell-cycle scoring was performed novel scoring algorithm https://carmonalab.github.io/UCell/UCell_Seurat_vignette.html
# Genesets used also in Seurat Cell-cycle scoring
seu.live &lt;- AddModuleScore_UCell(seu.live, assay = &quot;SCT&quot;,slot = &quot;data&quot;,features = cc.genes.updated.2019)

## Fetch data for plotting
data.CC.live &lt;- FetchData(seu.live, vars = c(&#39;G2M.Score&#39;, &#39;S.Score&#39;, &quot;Phase&quot;, &quot;donor&quot;, &quot;s.genes_UCell&quot;, &quot;g2m.genes_UCell&quot;))
data.CC.live$MKI67 &lt;- FetchData(seu.live[[&quot;SCT&quot;]], &quot;MKI67&quot;, slot = &quot;scale.data&quot;)$MKI67

## Plot CC score

p.CCscore.MKI67.live &lt;- ggplot(data = data.CC.live, aes(x = g2m.genes_UCell, y =s.genes_UCell , color = MKI67)) +
  geom_point( position = &#39;jitter&#39;, alpha = 0.8) +
  scale_color_gradientn(colors = c(&#39;lightgrey&#39;, &#39;dodgerblue2&#39;)) +
  geom_rect(mapping=aes(xmin=-0.01, xmax=0.1, ymin=-0.01, ymax=0.1), color=&quot;black&quot;, alpha=0)+
  cowplot::theme_cowplot() + 
  labs(title = &quot;RNA-based G2M and S-score mark \ncycling and non-cycling cells&quot;, x = &quot;G2M-score&quot;, y = &quot;S-score&quot;)+
  add.textsize+
  #facet_wrap(.~donor)+
  annotate(geom = &quot;text&quot;, label = &quot;Non-\ncycling \n~90%&quot;, x=0.01, y = 0.09,hjust = 0,vjust = 1, size = 2) +
  theme(legend.position = c(0.85,0.85), legend.key.size = unit(2, &#39;mm&#39;)) 

## Catagorize cells based on CCscore and MKI67 as cycling or non-cycling. 

data.CC.live &lt;- mutate(data.CC.live, Proliferation_state = ifelse(g2m.genes_UCell &gt;= 0.1 | s.genes_UCell &gt;=0.1 | MKI67 &gt;=1 , &quot;cycling&quot;,&quot;non-cycling&quot;))

seu.live &lt;- AddMetaData(seu.live, metadata = data.CC.live$Proliferation_state, col.name = &quot;Cellcycle_state&quot;)

Percentage_differentiated.live &lt;- data.CC.live %&gt;% dplyr::count(Proliferation_state = factor(Proliferation_state),group = factor(donor)) %&gt;% 
  group_by(group)%&gt;%
    mutate(pct = prop.table(n)) 

## Plot percentage cycling per donor
p.percentage.cyclinglive &lt;- ggplot(Percentage_differentiated.live, aes(x = Proliferation_state , y = pct, fill = group)) + #label = scales::percent(pct,accuracy = 0.1)
    geom_col(position = &#39;dodge&#39;) + 
    scale_y_continuous(labels = scales::percent) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;~90% non-cycling cells \nacross three donors&quot;,x=&quot;&quot;, y = &quot;Percentage (%)&quot;)+
scale_color_manual(&quot;Donor&quot;, values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
scale_fill_manual(&quot;Donor&quot;,values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  add.textsize+
  theme(legend.position = c(0.05,0.85), legend.key.size = unit(4, &#39;mm&#39;)) 


###          --- fix cells ------

# Cell-cycle scoring was performed novel scoring algorithm https://carmonalab.github.io/UCell/UCell_Seurat_vignette.html
# Genesets used also in Seurat Cell-cycle scoring
seu.fix &lt;- AddModuleScore_UCell(seu.fix, assay = &quot;SCT&quot;,slot = &quot;data&quot;,features = cc.genes.updated.2019)

## Fetch data for plotting
data.CC.fix &lt;- FetchData(seu.fix, vars = c(&#39;G2M.Score&#39;, &#39;S.Score&#39;, &quot;Phase&quot;, &quot;donor&quot;, &quot;s.genes_UCell&quot;, &quot;g2m.genes_UCell&quot;))
data.CC.fix$MKI67 &lt;- FetchData(seu.fix[[&quot;SCT&quot;]], &quot;MKI67&quot;, slot = &quot;scale.data&quot;)$MKI67
data.CC.fix$MKI67.PROT &lt;- FetchData(seu.fix[[&quot;PROT&quot;]], &quot;Ki67&quot;, slot = &quot;scale.data&quot;)$Ki67
data.CC.fix$`p-Rb` &lt;- FetchData(seu.fix[[&quot;PROT&quot;]], &quot;p-Rb&quot;, slot = &quot;scale.data&quot;)$`p-Rb`
data.CC.fix$CyclinA &lt;- FetchData(seu.fix[[&quot;PROT&quot;]], &quot;Cyclin A&quot;, slot = &quot;scale.data&quot;)$`Cyclin`
## Plot CC score

p.CCscore.MKI67.fix &lt;- ggplot(data = data.CC.fix, aes(x = g2m.genes_UCell, y =s.genes_UCell , color = MKI67)) +
  geom_point( position = &#39;jitter&#39;, alpha = 0.8) +
  scale_color_gradientn(colors = c(&#39;lightgrey&#39;, &#39;dodgerblue2&#39;)) +
  cowplot::theme_cowplot() + 
  labs(title = &quot;RNA-based G2M and S-score mark \ncycling and non-cycling cells&quot;, x = &quot;G2M-score&quot;, y = &quot;S-score&quot;)+
  add.textsize+
  #facet_wrap(.~donor)+
  geom_vline(xintercept = 0.1) +
  geom_hline(yintercept = 0.07) +
  annotate(geom = &quot;text&quot;, label = &quot;Non-\ncycling&quot;, x=0.01, y = 0.025, size = 2) +
  theme(legend.position = c(0.85,0.85), legend.key.size = unit(2, &#39;mm&#39;)) 

##The advantage of dataset is intracellular phospho-protein markers detected in protein dataset, which can be used to filter filter non-deviding cells. p-Rb and Cyclin A are used:

p.CCscore.pRb &lt;- ggplot(data = data.CC.fix, aes(x = `p-Rb`, y =CyclinA , color = MKI67.PROT)) +
  geom_point( position = &#39;jitter&#39;, alpha = 0.8) +
  scale_color_gradientn(colors = c(&#39;lightgrey&#39;, &#39;deeppink3&#39;)) +
  cowplot::theme_cowplot() + 
  labs(title = &quot;Intracellular protein markers classify fixed  \ndataset in cycling and non-cycling cells&quot;)+
  add.textsize+
  geom_rect(mapping=aes(xmin=-1.8, xmax=1.6, ymin=-2.8, ymax=2.3), color=&quot;black&quot;, alpha=0)+
  annotate(geom = &quot;text&quot;, label = &quot;Non-\ncycling \n&gt;95%&quot;, x=-1.7, y = 0.6,hjust = 0,vjust = 1, size = 2)+
  theme(legend.position = c(0.05,0.85), legend.key.size = unit(2, &#39;mm&#39;)) 


## Catagorize cells based on CCscore and MKI67 as cycling or non-cycling. 

data.CC.fix &lt;- mutate(data.CC.fix, Proliferation_state = ifelse( `p-Rb` &gt;= 1.6 | CyclinA &gt;= 2.3, &quot;cycling&quot;,&quot;non-cycling&quot;))


seu.fix &lt;- AddMetaData(seu.fix, metadata = data.CC.fix$Proliferation_state, col.name = &quot;Cellcycle_state&quot;)

Percentage_differentiated.fix &lt;- data.CC.fix %&gt;% dplyr::count(Proliferation_state = factor(Proliferation_state),group = factor(donor)) %&gt;% 
  group_by(group)%&gt;%
    mutate(pct = prop.table(n)) 

## Plot percentage cycling per donor
p.percentage.cycling.fix &lt;- ggplot(Percentage_differentiated.fix, aes(x = Proliferation_state , y = pct, fill = group, label = scales::percent(pct,accuracy = 0.1))) + 
    geom_col(position = &#39;dodge&#39;) + 
    geom_text(position = position_dodge(width = .9),    # move to center of bars
              vjust = -0.5,    # nudge above top of bar
              size = 1.8) + 
    scale_y_continuous(labels = scales::percent) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Fixed dataset classification per donor \n&gt;95% non-cycling cells&quot;,x=&quot;&quot;, y = &quot;Percentage (%)&quot;)+
scale_color_manual(&quot;Donor&quot;, values=c(&quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
scale_fill_manual(&quot;Donor&quot;,values=c(&quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  add.textsize+
  theme(legend.position = c(0.05,0.85), legend.key.size = unit(2, &#39;mm&#39;)) </code></pre>
</div>
<div id="differentiated-plasma-blastcells" class="section level2">
<h2>Differentiated Plasma-Blast/Cells</h2>
<p>To determine the cultures ‘differentiation state’ CD27-high &amp;
IgD-low cells are gated, representing differentiated
plasmablast/cells.</p>
<pre class="r"><code>data.CC.fix$CD27 &lt;- FetchData(seu.fix[[&quot;PROT&quot;]], &quot;CD27&quot;, slot = &quot;scale.data&quot;)$CD27
data.CC.fix$IgD &lt;- FetchData(seu.fix[[&quot;PROT&quot;]], &quot;IgD&quot;, slot = &quot;scale.data&quot;)$IgD

data.CC.live$CD27 &lt;- FetchData(seu.live[[&quot;PROT&quot;]], &quot;CD27&quot;, slot = &quot;scale.data&quot;)$CD27
data.CC.live$IgD &lt;- FetchData(seu.live[[&quot;PROT&quot;]], &quot;IgD&quot;, slot = &quot;scale.data&quot;)$IgD

## ------ live
p.scatter.CD27.IgD.live &lt;- ggplot(data = data.CC.live,aes(x = IgD, y = CD27, color = donor)) +
  geom_point( position = &#39;jitter&#39;, alpha = 0.8) +
  geom_rect(mapping=aes(xmin=-2.5, xmax=3, ymin=-2, ymax=2.8), color=&quot;black&quot;, alpha=0)+
  cowplot::theme_cowplot() + 
  labs(title = &quot;Protein-based gate of CD27+IgD- cells&quot;, x = &quot;IgD&quot;, y = &quot;CD27&quot;)+
  theme(legend.position = &quot;none&quot;) +
scale_color_manual(&quot;Donor&quot;, values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  annotate(geom = &quot;text&quot;, label = &quot;CD27+IgD- \n&gt;95%&quot;, x=-2.3, y = 2.5,hjust = 0,vjust = 1, size = 2)+
  add.textsize  

data.CC.live &lt;- mutate(data.CC.live, class_switch_plasma = ifelse(IgD &lt;= 3 &amp; CD27 &gt;= -2.5 , &quot;CD27+IgD-&quot;,&quot;other&quot;))
seu.live &lt;- AddMetaData(seu.live, metadata = data.CC.live$class_switch_plasma, col.name = &quot;CD27_IgD_state&quot;)

Percentage_class_switch_plasma.live &lt;- data.CC.live %&gt;% dplyr::count(class_switch_plasma = factor(class_switch_plasma),group = factor(donor)) %&gt;% 
  group_by(group)%&gt;%
    mutate(pct = prop.table(n)) 

p.percentage.class_switch_plasma.live &lt;- ggplot(Percentage_class_switch_plasma.live, aes(x = class_switch_plasma , y = pct, fill = group, label = scales::percent(pct, accuracy = 0.1))) + 
    geom_col(position = &#39;dodge&#39;) + 
    geom_text(position = position_dodge(width = .9),    # move to center of bars
              vjust = -0.5,    # nudge above top of bar
              size = 1.8) + 
    scale_y_continuous(labels = scales::percent) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;~95% cells in all three donors \nare Plasma Blast/Cells&quot;,x=&quot;&quot;, y = &quot;Percentage (%)&quot;)+
scale_fill_manual(name = &quot;Donor&quot;,values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;)) +
  theme(legend.position = c(0.7,0.85), legend.key.size = unit(2, &quot;mm&quot;))+
  add.textsize

## --- fix
## fix
p.scatter.CD27.IgD.fix &lt;- ggplot(data = data.CC.fix,aes(x = IgD, y = CD27, color = donor)) +
  geom_point( position = &#39;jitter&#39;, alpha = 0.8) +
  geom_rect(mapping=aes(xmin=-4, xmax=4.2, ymin=-1.3, ymax=2), color=&quot;black&quot;, alpha=0)+
  cowplot::theme_cowplot() + 
  labs(title = &quot;Protein-based gate of CD27+IgD- cells&quot;, x = &quot;IgD&quot;, y = &quot;CD27&quot;)+
theme(legend.position = c(0.85,0.85)) +
scale_color_manual(&quot;Donor&quot;, values=c( &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  add.textsize  +
  xlim(c(NA,7.5))

data.CC.fix &lt;- mutate(data.CC.fix, class_switch_plasma = ifelse(IgD &lt;= 4.2 &amp; CD27 &gt;= -1.3 , &quot;CD27+IgD-&quot;,&quot;other&quot;))
seu.fix &lt;- AddMetaData(seu.fix, metadata = data.CC.fix$class_switch_plasma, col.name = &quot;CD27_IgD_state&quot;)

Percentage_class_switch_plasma.fix &lt;- data.CC.fix %&gt;% dplyr::count(class_switch_plasma = factor(class_switch_plasma),group = factor(donor)) %&gt;% 
  group_by(group)%&gt;%
    mutate(pct = prop.table(n)) 

p.percentage.class_switch_plasma.fix &lt;- ggplot(Percentage_class_switch_plasma.fix, aes(x = class_switch_plasma , y = pct, fill = group, label = scales::percent(pct, accuracy = 0.1))) + 
    geom_col(position = &#39;dodge&#39;) + 
    geom_text(position = position_dodge(width = .9),    # move to center of bars
              vjust = -0.5,    # nudge above top of bar
              size = 1.8) + 
    scale_y_continuous(labels = scales::percent) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Fixed dataset % cells CD27+IgD-&quot;,x=&quot;&quot;, y = &quot;Percentage (%)&quot;)+
scale_fill_manual(name = &quot;Donor&quot;,values=c( &quot;#E69F00&quot;, &quot;#56B4E9&quot;)) +
  theme(legend.position = c(0.7,0.9), legend.key.size = unit(2, &#39;mm&#39;))+
  add.textsize</code></pre>
<pre class="r"><code>seu.live &lt;- SetIdent(seu.live,value = &quot;CD27_IgD_state&quot;)

## RNA differential expression
markers.CD27IgD.RNA &lt;- FindMarkers(seu.live, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;SCT&quot;, logfc.threshold = 0.01, test.use = &quot;wilcox&quot;, only.pos = T)
markers.CD27IgD.RNA &lt;- filter(markers.CD27IgD.RNA, p_val &lt;= 0.05)

p.CD27IgD.dotplot.genesign &lt;-DotPlot(seu.live,assay = &quot;SCT&quot;, features = rev(rownames(markers.CD27IgD.RNA)[1:20]), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -0.5, col.max = 0.5) +
  labs(x = &quot;Differentiation markers - RNA&quot;, y = &quot;&quot;) +
  cowplot::theme_cowplot() + 
  labs(title=paste0(&quot;Top 20 upregulated genes \n(&quot;, nrow(markers.CD27IgD.RNA),&quot; total)&quot;),x=&quot;Differential expressed genes \n(p-val &lt; 0.05, logfc &gt;= 0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D1 \n&quot;,&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D1 \n&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;)) 
#+  guides(size = &quot;none&quot;,color = &quot;none&quot;)

p.CD27IgD.Vln.gene.markers &lt;- VlnPlot(seu.live, assay=&quot;SCT&quot;, features = c(&quot;CD27&quot;,&quot;IRF4&quot;, &quot;MZB1&quot;, &quot;PTPRC&quot;, &quot;MS4A1&quot;), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, ncol =3, cols = c(colors.donors,colors.donors), combine = T, stack = T, flip = T) +
    add.textsize +
  theme(legend.position = &quot;&quot;, legend.key.size = unit(2, &#39;mm&#39;)) + 
  labs(title=&quot;CD27+IgD express plasmablast/cell \nRNA markers&quot;,x=&quot;&quot;, y = &quot;Expression&quot;)  +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))
 
## RNA differential expression
markers.CD27IgD.RNA.neg &lt;- FindMarkers(seu.live, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;SCT&quot;, logfc.threshold = -0.01, test.use = &quot;wilcox&quot;, only.pos = F)
markers.CD27IgD.RNA.neg &lt;- filter(markers.CD27IgD.RNA.neg, p_val &lt;= 0.05, avg_log2FC &lt; -0.01)

p.CD27IgD.dotplot.genesign.neg &lt;-DotPlot(seu.live,assay = &quot;SCT&quot;, features = rev(rownames(markers.CD27IgD.RNA.neg)[1:20]), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;size&quot;, col.min = -0.5, col.max = 0.5) +
  labs(x = &quot;Differentiation markers - RNA&quot;, y = &quot;&quot;) +
  cowplot::theme_cowplot() + 
  labs(title=paste0(&quot;Top 20 downregulated genes \n(&quot;, nrow(markers.CD27IgD.RNA.neg),&quot; total)&quot;),x=&quot;Differential expressed genes \n(p-val &lt; 0.05, logfc &lt;= -0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D1 \n&quot;,&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D1 \n&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;)) +
    guides(size =  guide_legend(title = &quot;Percent \nexpressed&quot;),color = guide_colorbar(title = &quot;Average scaled\nexpression&quot;))
#+  guides(size = &quot;none&quot;,color = &quot;none&quot;)

 
## Surface proteins
markers.CD27IgD.PROT &lt;- FindMarkers(seu.live, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;PROT&quot;, logfc.threshold = 0.01, test.use = &quot;wilcox&quot;, only.pos = T)
markers.CD27IgD.PROT &lt;- filter(markers.CD27IgD.PROT, p_val &lt;= 0.05)

p.CD27IgD.dotplot.PROTsign &lt;-DotPlot(seu.live,assay = &quot;PROT&quot;, features = rev(rownames(markers.CD27IgD.PROT)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -0.5, col.max = 0.5) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Differential upregulated \nSurface protein markers&quot;,x=&quot;Surface Proteins \n(p-val &lt; 0.005, logfc &gt;= 0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D1 \n&quot;,&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D1 \n&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;))+
  guides(size = &quot;none&quot;,color = &quot;none&quot;)

markers.CD27IgD.PROT.neg &lt;- FindMarkers(seu.live, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;PROT&quot;, logfc.threshold = 0.01, test.use = &quot;wilcox&quot;, only.pos = F)
markers.CD27IgD.PROT.neg &lt;- filter(markers.CD27IgD.PROT.neg, p_val &lt;= 0.005, avg_log2FC &lt; -0.01)

p.CD27IgD.dotplot.PROTsign.neg &lt;-DotPlot(seu.live,assay = &quot;PROT&quot;, features = rev(rownames(markers.CD27IgD.PROT.neg)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -0.5, col.max = 0.5) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Differential downregulated \nSurface protein markers&quot;,x=&quot;Surface Proteins \n(p-val &lt; 0.005, logfc &lt;= -0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D1 \n&quot;,&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D1 \n&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;))+
    guides(size =  &quot;none&quot;,color = guide_colorbar(title = &quot;Average scaled\nexpression&quot;))


## Intracellular proteins
seu.fix &lt;- SetIdent(seu.fix,value = &quot;CD27_IgD_state&quot;)

markers.CD27IgD.PROT.intra &lt;- FindMarkers(seu.fix, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;PROT&quot;, logfc.threshold = 0.05, test.use = &quot;wilcox&quot;, only.pos = T)
markers.CD27IgD.PROT.intra &lt;- filter(markers.CD27IgD.PROT.intra, p_val &lt;= 0.05)


p.CD27IgD.dotplot.PROT.intra.sign &lt;- DotPlot(seu.fix,assay = &quot;PROT&quot;, features = rev(rownames(markers.CD27IgD.PROT.intra)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -0.5, col.max = 0.5) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Differential upregulated \nIntracellular protein markers&quot;,x=&quot;Intracellular proteins \n(p-val &lt; 0.005, logfc &gt;= 0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;))

markers.CD27IgD.PROT.intra.neg &lt;- FindMarkers(seu.fix, ident.1 = &quot;CD27+IgD-&quot;, ident.2 = &quot;other&quot;, assay = &quot;PROT&quot;, logfc.threshold = 0.01, test.use = &quot;wilcox&quot;, only.pos = F)
markers.CD27IgD.PROT.intra.neg &lt;- filter(markers.CD27IgD.PROT.intra.neg, p_val &lt;= 0.005, avg_log2FC &lt; -0.01)

p.CD27IgD.dotplot.PROTsign.intra.neg &lt;-DotPlot(seu.fix,assay = &quot;PROT&quot;, features = rev(rownames(markers.CD27IgD.PROT.intra.neg)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -0.5, col.max = 0.5) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Differential downregulated \nSurface protein markers&quot;,x=&quot;Intracellular Proteins \n(p-val &lt; 0.005, logfc &lt;= -0.01)&quot;, y = &quot;&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D1 \n&quot;,&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D1 \n&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;))+
    guides(size =  &quot;none&quot;,color = guide_colorbar(title = &quot;Average scaled\nexpression&quot;))



markers.forviolin &lt;- c(&quot;CD138&quot;,&quot;BLIMP1&quot;,&quot;IRF4&quot;, &quot;IRF8&quot;, &quot;XBP1&quot;, &quot;CD27&quot;,&quot;CD24&quot;)[c(&quot;CD138&quot;,&quot;BLIMP1&quot;,&quot;IRF4&quot;, &quot;IRF8&quot;, &quot;XBP1&quot;, &quot;CD27&quot;,&quot;CD24&quot;) %in% rev(rownames(markers.CD27IgD.PROT.intra))]


p.CD27IgD.Vln.PROT.markers &lt;- VlnPlot(seu.fix, assay=&quot;PROT&quot;, features = rev(c(markers.forviolin)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, ncol =3, cols = c(colors.donors[2:3],colors.donors[2:3]), combine = T, stack = T, flip = T) +
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) + 
  labs(title=&quot;CD27+IgD express plasmablast/cell \nIntracellular protein markers&quot;,x=&quot;&quot;, y = &quot;Expression&quot;)  +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5))

p.CD27IgD.dotplot.PROT.intra.markers &lt;- DotPlot(seu.fix,assay = &quot;PROT&quot;, features = rev(c(markers.forviolin)), split.by = &quot;donor&quot;, group.by =  &quot;CD27_IgD_state&quot;, cols = &quot;RdBu&quot;,dot.scale = 3, scale.max = 100, scale.min = 0, scale = T, scale.by = &quot;radius&quot;, col.min = -1, col.max = 1) +
  labs(x = &quot;Differentiation markers - Surface Proteins&quot;, y = &quot;&quot;) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;CD27+IgD express plasmablast/cell \nIntracellular protein markers&quot;,x=&quot;Differentiation proteins&quot;, y = &quot;&quot;, colour = &quot;Legend Title\n&quot;)+
  coord_flip()+
    add.textsize +
  theme(legend.position = &quot;right&quot;, legend.key.size = unit(2, &#39;mm&#39;)) +
  scale_y_discrete(labels = c(&quot;D2 \nCD27+IgD-&quot;, &quot;D3&quot;,&quot;D2 \nother&quot;, &quot;D3&quot;)) +
  guides(size = &quot;none&quot;,color = guide_colorbar(title = &quot;Average \nExpression&quot;))</code></pre>
<div id="general-properties-seurat-datasets" class="section level4">
<h4>General properties seurat datasets:</h4>
<p>Fixed-cell dataset: (original and CD27+IgD- object )</p>
<pre class="r"><code>message(&quot;Fixed dataset:&quot;)
seu.fix</code></pre>
<pre><code>An object of class Seurat 
10114 features across 1038 samples within 3 assays 
Active assay: PROT (76 features, 76 variable features)
 2 other assays present: RNA, SCT</code></pre>
<pre class="r"><code>seu.fix.filtered &lt;- subset(seu.fix, Cellcycle_state == &quot;non-cycling&quot; &amp; CD27_IgD_state == &quot;CD27+IgD-&quot;)
message(&quot;CD27+IgD- object:&quot;)
seu.fix.filtered</code></pre>
<pre><code>An object of class Seurat 
10114 features across 940 samples within 3 assays 
Active assay: PROT (76 features, 76 variable features)
 2 other assays present: RNA, SCT</code></pre>
<p>Live-cell dataset: (original and CD27+IgD- object )</p>
<pre class="r"><code>seu.live</code></pre>
<pre><code>An object of class Seurat 
20366 features across 1433 samples within 3 assays 
Active assay: PROT (50 features, 50 variable features)
 2 other assays present: RNA, SCT</code></pre>
<pre class="r"><code>seu.live.filtered &lt;- subset(seu.live, Cellcycle_state == &quot;non-cycling&quot; &amp; CD27_IgD_state == &quot;CD27+IgD-&quot;)

seu.live.filtered</code></pre>
<pre><code>An object of class Seurat 
20366 features across 1231 samples within 3 assays 
Active assay: PROT (50 features, 50 variable features)
 2 other assays present: RNA, SCT</code></pre>
<pre class="r"><code>##Save seurat objects with all cells &amp; information on CC , CD27, and Ig-subclasses.

#saveRDS(seu.fix, &quot;output/seu.fix_norm_cellstate.rds&quot;)
#saveRDS(seu.live, &quot;output/seu.live_norm_cellstate.rds&quot;)

seu.fix &lt;- seu.fix.filtered
seu.live &lt;- seu.live.filtered</code></pre>
</div>
</div>
<div id="ig-classifaction" class="section level2">
<h2>Ig-classifaction</h2>
<div id="seurat-protein-based" class="section level3">
<h3>Seurat, protein-based</h3>
<p>The supernatant of differentiated cells shows excretion of IgA and
IgG antibodies, showing these cell-types should be within the population
at 11 days. Using protein-measurements from antibodies against the major
Ig-classes, and (in live-cell dataset) the Ig-genes, cells are clustered
and represented in a UMAP.</p>
<pre class="r"><code>## Run PCA on Ig proteins
seu.fix &lt;- RunPCA(seu.fix, reduction.name = &#39;pcaIG&#39;, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;))

seu.fix &lt;- RunUMAP(seu.fix,dims = 1:4, reduction = &quot;pcaIG&quot;, reduction.name = &quot;IGUMAP&quot;)
seu.fix &lt;- FindNeighbors(seu.fix, reduction = &quot;pcaIG&quot;,assay = &quot;PROT&quot;, dims = 1:4, graph.name = &quot;pcaIG_nn&quot;)
seu.fix &lt;- FindClusters(seu.fix, graph.name = &quot;pcaIG_nn&quot;,  algorithm = 3,resolution = 0.2, verbose = FALSE)
seu.fix[[&quot;clusters_pcaIG&quot;]] &lt;- Idents(object = seu.fix)


seu.fix &lt;- RenameIdents(object = seu.fix, `0` = &quot;IgM&quot;, `1` = &quot;IgG&quot;, `2` = &quot;IgA&quot;)
seu.fix[[&quot;clusters_pcaIG_named&quot;]] &lt;- Idents(object = seu.fix)

## ---- live cells

## Run PCA on Ig proteins and on Ig genes
seu.live &lt;- RunPCA(seu.live, assay = &quot;PROT&quot;,reduction.name = &#39;pcaIGPROT&#39;, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;))

seu.live &lt;- RunUMAP(seu.live,dims = 1:4, reduction = &quot;pcaIGPROT&quot;, reduction.name = &quot;IGUMAP&quot;)
seu.live &lt;- FindNeighbors(seu.live, reduction = &quot;pcaIGPROT&quot;,assay = &quot;PROT&quot;, dims = 1:4, graph.name = &quot;pcaIG_nn&quot;)
seu.live &lt;- FindClusters(seu.live, graph.name = &quot;pcaIG_nn&quot;,  algorithm = 3,resolution = 0.2, verbose = FALSE)
seu.live[[&quot;clusters_pcaIG&quot;]] &lt;- Idents(object = seu.live)


seu.live &lt;- RenameIdents(object = seu.live, `0` = &quot;IgM&quot;, `1` = &quot;IgG&quot;, `2` = &quot;IgA&quot;)
seu.live[[&quot;clusters_pcaIG_named&quot;]] &lt;- Idents(object = seu.live)</code></pre>
<pre class="r"><code>### Plots Ig-protein based clustering
p.umap.Ig.prot.fix &lt;- FeaturePlot(seu.fix, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;),slot = &quot;scale.data&quot;, reduction = &#39;IGUMAP&#39;, max.cutoff = 2, 
                  cols = c(&quot;lightgrey&quot;,&quot;deeppink3&quot;), ncol = 3, pt.size = 0.9)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.85), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))

p.clusters.umap.fix &lt;- DimPlot(seu.fix, reduction = &#39;IGUMAP&#39;,group.by =&quot;clusters_pcaIG_named&quot; , label = TRUE, repel = TRUE, label.size = 2.5, cols = c(&quot;#009E73&quot;, &quot;#D55E00&quot;,&quot;#F0E442&quot;)
)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Clusters&quot;, title = &quot;Fixed cells clusters based on \nIg-protein marker expression&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))

p.umap.Ig.prot.live &lt;- FeaturePlot(seu.live, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;),slot = &quot;scale.data&quot;, reduction = &#39;IGUMAP&#39;, max.cutoff = 2, 
                  cols = c(&quot;lightgrey&quot;,&quot;deeppink3&quot;), ncol = 3, pt.size = 0.5)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))

p.umap.Ig.RNA.live &lt;- FeaturePlot(seu.live, features = c(&quot;IGHM&quot;,&quot;IGHA1&quot;,&quot;IGHG1&quot;),slot = &quot;scale.data&quot;, reduction = &#39;IGUMAP&#39;, max.cutoff = 3, 
                  cols = c(&quot;lightgrey&quot;,&quot;dodgerblue3&quot;), ncol = 3, pt.size =  0.5)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))
          
p.clusters.umap.live &lt;- DimPlot(seu.live, reduction = &#39;IGUMAP&#39;, label = TRUE, repel = TRUE, label.size = 2.5,pt.size =  0.5, cols = c(&quot;#009E73&quot;, &quot;#D55E00&quot;,&quot;#F0E442&quot;,&quot;grey&quot;)
)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Clusters&quot;, title = &quot;Igsubtype classification based on \nProtein modality&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(1, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))

p.donor.umap.live &lt;- DimPlot(seu.live, reduction = &#39;IGUMAP&#39;, group.by = &quot;donor&quot;,label = F, pt.size =  0.5,repel = TRUE, label.size = 2.5, cols = colors.donors)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Donor&quot;, title = &quot;Ig-subtypes classified across \nthree donors&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85, 0.95), legend.key.size = unit(1, &#39;mm&#39;), legend.text =element_text(size=6), legend.title =element_text(size=7)  ) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))</code></pre>
<pre class="r"><code>fix.meta &lt;- seu.fix.filtered@meta.data %&gt;%
  mutate(cell = rownames(seu.fix.filtered@meta.data),
         dataset = &quot;fix&quot;) 
live.meta &lt;- seu.live.filtered@meta.data%&gt;%
  mutate(cell = rownames(seu.live.filtered@meta.data),
         dataset = &quot;live&quot;)</code></pre>
</div>
<div id="mofa-analysis" class="section level3">
<h3>MOFA analysis</h3>
<pre class="r"><code>library(MOFA2)</code></pre>
<pre class="r"><code>fix.DT.RNA &lt;- seu.fix.filtered@assays$SCT@scale.data[rownames(seu.fix.filtered@assays$SCT@scale.data) %chin% VariableFeatures(seu.fix.filtered,assay = &quot;SCT&quot;),]
fix.var.genes &lt;- rownames(fix.DT.RNA)

fix.DT.PROT &lt;- seu.fix.filtered@assays$PROT@scale.data[rownames(seu.fix.filtered@assays$PROT@scale.data) %chin% VariableFeatures(seu.fix.filtered,assay = &quot;PROT&quot;),]
fix.var.prot &lt;- rownames(fix.DT.PROT)

live.DT.RNA &lt;- seu.live.filtered@assays$SCT@scale.data[rownames(seu.live.filtered@assays$SCT@scale.data) %chin% VariableFeatures(seu.live.filtered,assay = &quot;SCT&quot;),]
live.var.genes &lt;- rownames(live.DT.RNA)

live.DT.PROT &lt;- seu.live.filtered@assays$PROT@scale.data[rownames(seu.live.filtered@assays$PROT@scale.data) %chin% VariableFeatures(seu.live.filtered,assay = &quot;PROT&quot;),]
live.var.prot &lt;- rownames(live.DT.PROT)

fix.meta &lt;- seu.fix.filtered@meta.data %&gt;%
  mutate(cell = rownames(seu.fix.filtered@meta.data),
         dataset = &quot;fix&quot;) 
live.meta &lt;- seu.live.filtered@meta.data%&gt;%
  mutate(cell = rownames(seu.live.filtered@meta.data),
         dataset = &quot;live&quot;)

## 
common.proteins &lt;- c(&quot;TACI&quot;,&quot;CD70&quot;,&quot;IgD&quot;,&quot;CD86&quot;,&quot;CD5&quot;,&quot;PDL1&quot;,&quot;PDL2&quot;,&quot;CD21&quot;,&quot;IgM&quot;,&quot;CD11b&quot;,&quot;CD53&quot;,&quot;CD45&quot;,&quot;CD80&quot;,&quot;IgG&quot;,&quot;IgA&quot;,&quot;IgE&quot;,&quot;BAFFR&quot;)

fix.DT.RNA &lt;- data.table(fix.DT.RNA) %&gt;%
  mutate(gene = fix.var.genes)

fix.DT.PROT &lt;- data.table(fix.DT.PROT) %&gt;%
  mutate(gene = fix.var.prot)
fix.DT.PROT.unique &lt;- fix.DT.PROT[!(fix.var.prot %chin% common.proteins),] %&gt;%
  mutate(gene = paste0(gene, &quot;_f&quot;))

fix.DT.PROT.common &lt;- fix.DT.PROT[(fix.var.prot %chin% common.proteins),]
fix.DT.PROT.rename &lt;- bind_rows(fix.DT.PROT.unique,fix.DT.PROT.common)
#rownames(fix.DT.PROT.rename) &lt;- fix.DT.PROT.rename$gene

live.DT.RNA &lt;- data.table(live.DT.RNA) %&gt;%
  mutate(gene = live.var.genes)

live.DT.PROT &lt;- data.table(live.DT.PROT) %&gt;%
  mutate(gene = live.var.prot)
live.DT.PROT.unique &lt;- live.DT.PROT[!(live.var.prot %chin% common.proteins),] %&gt;%
  mutate(gene = paste0(gene, &quot;_l&quot;))
live.DT.PROT.common &lt;- live.DT.PROT[(live.var.prot %chin% common.proteins),]
live.DT.PROT.rename &lt;- bind_rows(live.DT.PROT.unique,live.DT.PROT.common)
## merge data
meta.all &lt;- merge.data.table(x = fix.meta, y=live.meta, all = TRUE)
meta.all$sample &lt;- meta.all$cell
meta.all$group &lt;- meta.all$donor

RNA.all &lt;- merge.data.table(x = fix.DT.RNA, y=live.DT.RNA, all = TRUE)
rownames(RNA.all) &lt;- RNA.all$gene
RNA.all &lt;- RNA.all[,gene:=NULL]

PROT.all &lt;- merge.data.table(x = fix.DT.PROT.rename, y=live.DT.PROT.rename, all = TRUE)

rownames(PROT.all) &lt;- PROT.all$gene
PROT.all &lt;- PROT.all[,gene:=NULL]

RNA.live.only &lt;- RNA.all[rownames(RNA.all) %chin% VariableFeatures(seu.live.filtered, assay = &quot;SCT&quot;),]
genes.RNA.live.only &lt;- rownames(RNA.all)[rownames(RNA.all) %chin% VariableFeatures(seu.live.filtered, assay = &quot;SCT&quot;)]
RNA.live.only &lt;- as.matrix(RNA.live.only)
#genes.RNA.live.only &lt;- tolower(genes.RNA.live.only)
rownames(RNA.live.only) &lt;- genes.RNA.live.only
#RNA.live.only[,c(which(meta.all$dataset== &quot;fix&quot;))] &lt;- NA #?

## split data tables

### PROT

PROT.common &lt;- PROT.all[rownames(PROT.all) %chin% common.proteins,]
list.common.prot &lt;- rownames(PROT.all)[rownames(PROT.all) %chin% common.proteins]
PROT.common &lt;- as.matrix(PROT.common, rownames.value    = list.common.prot)
#rownames(PROT.common) &lt;- paste0(rownames(PROT.common),&quot;_PROT&quot;)
#PROT.IG &lt;- PROT.all[rownames(PROT.all) %chin% c(&quot;IgA&quot;, &quot;IgM&quot;, &quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;),]
#list.IG.prot &lt;- rownames(PROT.all)[rownames(PROT.all) %chin% c(&quot;IgA&quot;, &quot;IgM&quot;, &quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;)]
#PROT.IG &lt;- as.matrix(PROT.IG, rownames.value   = list.IG.prot)

PROT.fix &lt;- PROT.all[c(rownames(PROT.all) %chin% fix.DT.PROT.unique$gene),]
list.fix.prot &lt;-rownames(PROT.all)[c(rownames(PROT.all) %chin% fix.DT.PROT.unique$gene)]
PROT.fix &lt;- as.matrix(PROT.fix, rownames.value  = list.fix.prot)
#rownames(PROT.fix) &lt;- paste0(rownames(PROT.fix),&quot;_PROT&quot;)

PROT.live &lt;- PROT.all[c(rownames(PROT.all) %chin% live.DT.PROT.unique$gene) ,]
list.live.prot &lt;-  rownames(PROT.all)[c(rownames(PROT.all) %chin% live.DT.PROT.unique$gene) ]
PROT.live &lt;- as.matrix(PROT.live, rownames.value    = list.live.prot)


### RNA 

RNA.common &lt;- RNA.all[rownames(RNA.all) %chin% rownames(seu.fix.filtered[[&quot;SCT&quot;]])[rownames(seu.fix.filtered[[&quot;SCT&quot;]]) %chin% rownames(seu.live.filtered[[&quot;SCT&quot;]])],]
list.common.RNA &lt;- rownames(RNA.all)[rownames(RNA.all) %chin% rownames(seu.fix.filtered[[&quot;SCT&quot;]])[rownames(seu.fix.filtered[[&quot;SCT&quot;]]) %chin% rownames(seu.live.filtered[[&quot;SCT&quot;]])]]
list.common.RNA &lt;- tolower(list.common.RNA)
RNA.common &lt;- as.matrix(RNA.common, rownames.value  = list.common.RNA)

#RNA.IG &lt;- RNA.all[rownames(RNA.all) %chin% c(&quot;IgA&quot;, &quot;IgM&quot;, &quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;),]
#list.IG.RNA &lt;- rownames(RNA.all)[rownames(RNA.all) %chin% c(&quot;IgA&quot;, &quot;IgM&quot;, &quot;IgG&quot;, &quot;IgD&quot;, &quot;IgE&quot;)]
#RNA.IG &lt;- as.matrix(RNA.IG, rownames.value = list.IG.RNA)

RNA.fix &lt;- RNA.all[!c(rownames(RNA.all) %chin% list.common.RNA) &amp; rownames(RNA.all) %chin% rownames(seu.fix.filtered[[&quot;SCT&quot;]]),]
list.fix.RNA &lt;-rownames(RNA.all)[!c(rownames(RNA.all) %chin% list.common.RNA) &amp; rownames(RNA.all) %chin% rownames(seu.fix.filtered[[&quot;SCT&quot;]])]
list.fix.RNA &lt;- tolower(list.fix.RNA)
RNA.fix &lt;- as.matrix(RNA.fix, rownames.value    = list.fix.RNA)

RNA.live &lt;- RNA.all[!c(rownames(RNA.all) %chin% list.common.RNA) &amp; !c(rownames(RNA.all) %chin% list.fix.RNA) ,]
list.live.RNA &lt;-  rownames(RNA.all)[!c(rownames(RNA.all) %chin% list.common.RNA) &amp; !c(rownames(RNA.all) %chin% list.fix.RNA) ]
#list.live.RNA &lt;- tolower(list.live.RNA)
RNA.live &lt;- as.matrix(RNA.live, rownames.value  = list.live.RNA)

###
all.features &lt;- c(rownames(RNA.live.only),rownames(PROT.common), rownames(PROT.live), rownames(PROT.fix))
duplicated.features &lt;- all.features[duplicated(all.features)]

RNA.live.only &lt;- data.table(RNA.live.only) %&gt;%
  mutate(gene = rownames(RNA.live.only)) 

rownames.rna.live.only &lt;- RNA.live.only$gene
RNA.live.only &lt;- RNA.live.only %&gt;%
  dplyr::select(-c(&quot;gene&quot;))
RNA.live.only &lt;- as.matrix(RNA.live.only)
#genes.RNA.live.only &lt;- tolower(genes.RNA.live.only)
rownames(RNA.live.only) &lt;- rownames.rna.live.only

all.features &lt;- c(rownames(RNA.live.only),rownames(PROT.common), rownames(PROT.live), rownames(PROT.fix))
duplicated.features &lt;- all.features[duplicated(all.features)]</code></pre>
<pre class="r"><code>myfiles &lt;- list.files(path=&quot;output/&quot;, pattern = &quot;.rds$&quot;)

if(&quot;MOFA_analysis_Donorgroup.rds&quot; %in%  myfiles){mofa &lt;- readRDS(&quot;output/MOFA_analysis_Donorgroup.rds&quot;)} else { #If so, read object, else do:
     
      mofa &lt;- create_mofa(data = list(RNA = RNA.live.only,  PROT.common = PROT.common,PROT.live = PROT.live, PROT.fix = PROT.fix ), groups = meta.all$group)

      # Default settings used (try 15 factors, excludes all non-informative factors)
      data_opts &lt;- get_default_data_options(mofa)
      model_opts &lt;- get_default_model_options(mofa)
      train_opts &lt;- get_default_training_options(mofa)
      train_opts$seed &lt;- 42 # use same seed for reproducibility
      mofa &lt;- prepare_mofa(
        object = mofa,
        data_options = data_opts,
        model_options = model_opts,
        training_options = train_opts
        )

    mofa &lt;- run_mofa(mofa, outfile = &quot;output/MOFA_analysis_Donorgroup.hdf5&quot;, use_basilisk = TRUE)
    mofa &lt;- run_umap(mofa)
    samples_metadata(mofa) &lt;- meta.all
    #saveRDS(mofa, file= &quot;output/MOFA_analysis_Donorgroup.rds&quot;)
  
}
mofa</code></pre>
<pre><code>Trained MOFA with the following characteristics: 
 Number of views: 4 
 Views names: RNA PROT.common PROT.live PROT.fix 
 Number of features (per view): 3000 17 33 59 
 Number of groups: 3 
 Groups names: D1 D2 D3 
 Number of samples (per group): 433 871 867 
 Number of factors: 9 </code></pre>
<div id="mofa-description" class="section level4">
<h4>MOFA description</h4>
<pre class="r"><code>p.mofaoverview.input &lt;- plot_data_overview(mofa)

## Variance per factor
p.variance.perfactor.all &lt;- plot_variance_explained(mofa, x=&quot;view&quot;, y=&quot;factor&quot;) +
    labs(title = &quot;Variance explained by each factor per modality&quot;)  +
  RotatedAxis()

## variance total
p.variance.total &lt;- plot_variance_explained(mofa, x=&quot;view&quot;, y=&quot;factor&quot;, plot_total = T)
# 
p.variance.total &lt;- plot_variance_explained(mofa, x=&quot;group&quot;, y=&quot;factor&quot;, plot_total = T)
p.variance.total &lt;- p.variance.total[[2]] +
  add.textsize +
  labs(title = &quot;Total variance per modality&quot;) +
  geom_text(aes(label=round(R2,1)), vjust=1.6, color=&quot;white&quot;, size=2.5) +
  theme_classic()+
  RotatedAxis()

factors.selected &lt;- paste0(&quot;Factor&quot;,1:5)

plot.rank.PROT &lt;- plot_weights(mofa, 
  view = &quot;PROT.common&quot;, 
  factors = factors.selected, 
  nfeatures = 10, 
  text_size =3
) +   
  labs(title = &quot;Top 10  Common Protein loadings per factor&quot;) +
  theme(axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
)</code></pre>
<pre class="r"><code>p.factors.Igclasses.PROT&lt;- plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;IgM&quot;,dot_size = 1.2
) +
  labs(title=&quot;MOFA factor 1 &amp; 2 distinguish Ig-classes \nProtein expression&quot;)+
  plot_factors(mofa, 
  factors =c(1:2),
  color_by = &quot;IgA&quot;,dot_size = 1.2
)+
  plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;IgG&quot;,dot_size = 1.2
) &amp;
    add.textsize &amp;  
  labs(x = &quot;Factor 1&quot;, y = &quot;Factor 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0)) &amp;
  scale_fill_gradient(low = &quot;white&quot;, high = &quot;deeppink3&quot;)

p.factors.Igclasses.mRNA &lt;- plot_factors(mofa, 
  factors =c(1:2),
  color_by = &quot;IGHM&quot;,dot_size = 1.2, show_missing = F,alpha = 0.5
) +labs(title=&quot;mRNA expression&quot;)+
  plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;IGHA1&quot;,dot_size = 1.2, show_missing = F,alpha = 0.5
) +
  plot_factors(mofa, 
  factors =c(1:2),
  color_by = &quot;IGHG1&quot;,dot_size = 1.2, show_missing = F,alpha = 0.5
) &amp;
    add.textsize &amp;  
  labs(x = &quot;Factor 1&quot;, y = &quot;Factor 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0))  &amp;
  scale_fill_gradient(low = &quot;white&quot;, high = &quot;dodgerblue3&quot;)</code></pre>
</div>
<div id="clustering-ig-classes" class="section level4">
<h4>Clustering Ig-classes</h4>
<pre class="r"><code>clusters &lt;- cluster_samples(mofa, k=3, factors=c(1,2) )

meta.all.cluster &lt;-  meta.all[,] %&gt;% 
  left_join(data.frame(sample = names(clusters$cluster), cluster = clusters$cluster))
meta.all.cluster$cluster &lt;- as.factor(meta.all.cluster$cluster)
samples_metadata(mofa) &lt;-meta.all.cluster</code></pre>
<pre class="r"><code>clustermeta &lt;- mofa@samples_metadata %&gt;%
  mutate(IgClass = ifelse(cluster == &quot;1&quot;, &quot;IgM&quot;, ifelse(cluster == &quot;3&quot;, &quot;IgA&quot;, &quot;IgG&quot;)))

samples_metadata(mofa) &lt;- clustermeta

Percentage_cluster &lt;- clustermeta %&gt;% dplyr::count(IgClass = factor(IgClass),group = factor(donor)) %&gt;% 
  group_by(group)%&gt;%
    mutate(pct = prop.table(n)) 

p.percentages.Igclasses.MOFAclusters &lt;- ggplot(Percentage_cluster, aes(x = IgClass , y = pct, fill = group, label = scales::percent(pct,accuracy = 0.1))) + 
    geom_col(position = &#39;dodge&#39;) + 
    geom_text(position = position_dodge(width = .9),    # move to center of bars
              vjust = -0.5,    # nudge above top of bar
              size = 1.8) + 
    scale_y_continuous(labels = scales::percent) +
  cowplot::theme_cowplot() + 
  labs(title=&quot;Percentage Ig-class&quot;,x=&quot;&quot;, y = &quot;Percentage (%)&quot;)+
  scale_x_discrete(labels = c(&quot;1&quot; = &quot;IgG&quot;, &quot;2&quot; ,&quot;3&quot;)) +
scale_color_manual(&quot;Donor&quot;, values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
scale_fill_manual(&quot;Donor&quot;,values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  add.textsize</code></pre>
<pre class="r"><code>p.MOFA.factors.dataset &lt;- plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;dataset&quot;,
  alpha = 0.8,
  dot_size = 1.2
) +
  labs(title = &quot;MOFA analysis integrates \nthe two datasets&quot;) +
scale_fill_manual(&quot;Dataset&quot;,values=c(&quot;lightgrey&quot;, &quot;black&quot;)) &amp;
    add.textsize &amp;  
  labs(x = &quot;Factor 1&quot;, y = &quot;Factor 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0)) 
  
p.MOFA.factors.donors &lt;- plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;donor&quot;,
  alpha = 0.8,
  dot_size = 1.2
) +labs(title = &quot;MOFA factors are computed \nacross three donors&quot;) +
scale_fill_manual(&quot;Donor&quot;,values=colors.donors) &amp;
    add.textsize &amp;  
  labs(x = &quot;Factor 1&quot;, y = &quot;Factor 2&quot;, color = &quot;Donor&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0)) 


p.MOFA.factors.cluster &lt;-  plot_factors(mofa, 
   factors =c(1:2),
  color_by = &quot;IgClass&quot;,
  dot_size = 1.2
) +labs(title = &quot;k-means clustering on Factor 1 &amp; 2&quot;) +
scale_fill_manual(&quot;Cluster&quot;,values= colors.clusters) &amp;
  labs(x = &quot;Factor 1&quot;, y = &quot;Factor 2&quot;, color = &quot;Donor&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=10), axis.ticks = element_blank(),
          axis.text=element_text(size=10),
          plot.title = element_text(size=10, face = &quot;bold&quot;,hjust = 0)) </code></pre>
<pre class="r"><code>p.loadings.scatter &lt;- plot_weights_scatter(mofa, factors = c(1,2), view = &quot;PROT.common&quot;)+
  ggrepel::geom_text_repel(aes(x=x, y=y, label=feature), size=2.5, max.overlaps = 10)+
   coord_cartesian(xlim =c(-0.5,0.5), ylim = c(-0.5,0.5)) +

plot_weights_scatter(mofa, factors = c(1,2), view = &quot;PROT.fix&quot;)+
  ggrepel::geom_text_repel(aes(x=x, y=y, label=feature), size=2.5, max.overlaps = 10)+
   coord_cartesian(xlim =c(-0.6,0.6), ylim = c(-0.6,0.6)) +

plot_weights_scatter(mofa, factors = c(1,2), view = &quot;PROT.live&quot;)+
  ggrepel::geom_text_repel(aes(x=x, y=y, label=feature), size=2.5, max.overlaps = 10) +

plot_weights_scatter(mofa, factors = c(1,2), view = &quot;RNA&quot;)+
  ggrepel::geom_text_repel(aes(x=x, y=y, label=feature), size=2, max.overlaps =20) +
   coord_cartesian(xlim =c(-0.002,0.002), ylim = c(-0.002,0.002)) &amp;
  add.textsize

p.topweights.positive &lt;- plot_top_weights(mofa, factors = c(1,2), sign = &quot;positive&quot;, view = &quot;RNA&quot;, nfeatures = 35) +
  plot_top_weights(mofa, factors = c(1,2), sign = &quot;positive&quot;, view = &quot;PROT.common&quot;) /
  plot_top_weights(mofa, factors =c(1,2), sign = &quot;positive&quot;, view = &quot;PROT.fix&quot;)/
  plot_top_weights(mofa, factors = c(1,2), sign = &quot;positive&quot;, view = &quot;PROT.live&quot;) &amp;
  add.textsize

p.topweights.negative &lt;- plot_top_weights(mofa, factors = c(1,2), sign = &quot;negative&quot;, view = &quot;RNA&quot;, nfeatures = 35) +
  plot_top_weights(mofa, factors = c(1,2), sign = &quot;negative&quot;, view = &quot;PROT.common&quot;) /
  plot_top_weights(mofa, factors = c(1,2), sign = &quot;negative&quot;, view = &quot;PROT.fix&quot;)/
  plot_top_weights(mofa, factors = c(1,2), sign = &quot;negative&quot;, view = &quot;PROT.live&quot;) &amp;
  add.textsize

p.MOFA.factors.clusters &lt;- plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;dataset&quot;,
  alpha = 0.5,
  dot_size = 1.2
) +
  scale_color_manual(&quot;Dataset&quot;, values=c(&quot;yellow&quot;, &quot;blue&quot;))+
  labs(title = &quot;MOFA factors integrate two \n datasets across donors&quot;) +
scale_fill_manual(&quot;Donor&quot;,values=c(&quot;yellow&quot;, &quot;blue&quot;)) +
  
plot_factors(mofa, 
  factors = c(1:2),
  color_by = &quot;donor&quot;,
  alpha = 0.8,
  dot_size = 1.2
) +scale_color_manual(&quot;Donor&quot;, values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;))+
  labs(title = &quot;Donors overlab (no batch/donor effects)&quot;) +
scale_fill_manual(&quot;Donor&quot;,values=c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;)) +
  plot_factors(mofa, 
   factors =c(1:2),
  color_by = &quot;IgClass&quot;,
  dot_size = 1.2
) +
  labs(title = &quot;K-means clustering on Factor 1&amp;2&quot;)+ scale_color_manual(&quot;Ig-class&quot;, values=c(&quot;#009E73&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;))+
scale_fill_manual(&quot;Ig-class&quot;,values=c(c(&quot;#009E73&quot;, &quot;#D55E00&quot;, &quot;#CC79A7&quot;))) &amp; add.textsize</code></pre>
</div>
<div id="add-clusters-factors-to-seurat-object" class="section level4">
<h4>Add clusters &amp; Factors to Seurat object</h4>
<pre class="r"><code>fix.meta2 &lt;- fix.meta %&gt;%
  mutate(sample_cell = rownames(fix.meta))

clustermeta2 &lt;- clustermeta %&gt;%
  mutate(sample_cell = sample) %&gt;%
  dplyr::select(c(sample_cell,IgClass, clusterMOFA = cluster))

factorvalues &lt;- get_factors(mofa,factors = c(1:2),as.data.frame = TRUE )

factorvalues &lt;- factorvalues %&gt;%
  dplyr::select(-group)%&gt;%
  spread(factor,value)%&gt;%
  mutate(sample_cell = sample) %&gt;%
  dplyr::select(c(sample_cell,Factor1, Factor2))

fix.meta.new &lt;- left_join(fix.meta2, clustermeta2)
fix.meta.new &lt;- left_join(fix.meta.new,factorvalues)
rownames(fix.meta.new) &lt;- fix.meta.new$sample_cell
seu.fix &lt;- AddMetaData(seu.fix, fix.meta.new)
seu.fix &lt;- SetIdent(seu.fix,value = &quot;IgClass&quot;)
Idents(seu.fix) &lt;- factor(x = Idents(seu.fix), levels = c(&quot;IgM&quot;, &quot;IgG&quot;, &quot;IgA&quot;))</code></pre>
<pre class="r"><code>live.meta2 &lt;- live.meta %&gt;%
  mutate(sample_cell = rownames(live.meta))

clustermeta2 &lt;- clustermeta %&gt;%
  mutate(sample_cell = sample) %&gt;%
  dplyr::select(c(sample_cell,IgClass))

factorvalues &lt;- get_factors(mofa,factors = c(1:2),as.data.frame = TRUE )

factorvalues &lt;- factorvalues %&gt;%
  dplyr::select(-group)%&gt;%
  spread(factor,value)%&gt;%
  mutate(sample_cell = sample) %&gt;%
  dplyr::select(c(sample_cell,Factor1, Factor2))

live.meta.new &lt;- left_join(live.meta2, clustermeta2)
live.meta.new &lt;- left_join(live.meta.new,factorvalues)
rownames(live.meta.new) &lt;- live.meta.new$sample_cell
seu.live &lt;- AddMetaData(seu.live, live.meta.new)
seu.live &lt;- SetIdent(seu.live,value = &quot;IgClass&quot;)
Idents(seu.live) &lt;- factor(x = Idents(seu.live), levels = c(&quot;IgM&quot;, &quot;IgG&quot;, &quot;IgA&quot;))</code></pre>
</div>
</div>
</div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<pre class="r"><code>fig.1.row1 &lt;- plot_grid(p.CCscore.MKI67.live, 
                        p.CCscore.pRb,
                        p.percentage.cyclinglive,
                     labels = panellabels[4:6], label_size = 10, ncol = 3, rel_widths = c(1.2,1.2,0.9))

fig.1.row2 &lt;- plot_grid(p.scatter.CD27.IgD.live,
                        p.CD27IgD.dotplot.PROT.intra.markers,
                        p.CD27IgD.Vln.gene.markers,
                     labels = panellabels[7:9], label_size = 10, ncol = 3, rel_widths = c(1.2,1.2,0.9))


Fig.1.full &lt;- plot_grid(fig.1.row1, fig.1.row2, labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1,1))

# #ggsave(Fig.1.full, filename = &quot;output/paper_figures/Fig1D-I.eps&quot;, width = 177, height = 120, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
# #ggsave(Fig.1.full, filename = &quot;output/paper_figures/Fig1D-I.TIFF&quot;, width = 177, height = 120, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>Fig.1.full</code></pre>
<p><img src="figure/cellstate.Rmd/Figure1-1.png" width="691.2" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<pre class="r"><code>fig.2.row1 &lt;- plot_grid( p.factors.Igclasses.PROT,p.MOFA.factors.dataset, labels = panellabels[c(1,2)], label_size = 10, rel_widths = c(2.1,0.9))

fig.2.row2 &lt;- plot_grid( p.factors.Igclasses.mRNA,p.MOFA.factors.donors,labels = c(&quot;&quot;, panellabels[3]), label_size = 10,
          rel_widths = c(2.1,0.9))


fig.2.full &lt;- plot_grid(fig.2.row1, fig.2.row2, labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1,1))

# #ggsave(fig.2.full, filename = &quot;output/paper_figures/Fig2.eps&quot;, width = 177, height = 120, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
# #ggsave(fig.2.full, filename = &quot;output/paper_figures/Fig2.TIFF&quot;, width = 177, height = 120, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>fig.2.full</code></pre>
<p><img src="figure/cellstate.Rmd/fig.2.full-1.png" width="691.2" style="display: block; margin: auto;" /></p>
</div>
<div id="supplementary-figures" class="section level2">
<h2>Supplementary figures</h2>
<div id="supplement-cd27igd--subclasses" class="section level3">
<h3>Supplement CD27+IgD- subclasses</h3>
<pre class="r"><code>p.percentages &lt;- plot_grid(p.percentage.cycling.fix,
          p.percentage.class_switch_plasma.live,
          p.percentage.class_switch_plasma.fix,
          labels = panellabels[1:3], label_size = 10, ncol = 3, rel_widths = c(1,1))

p.dotplot.genes&lt;- plot_grid(p.CD27IgD.dotplot.genesign,
          p.CD27IgD.dotplot.genesign.neg,
          labels = c(panellabels[4:5]), label_size = 10, ncol = 2, rel_heights =  c(1,1))


p.dotplot.proteins.surface &lt;- plot_grid(p.CD27IgD.dotplot.PROTsign,
          p.CD27IgD.dotplot.PROTsign.neg,
          labels = c(panellabels[6:7]), label_size = 10, ncol = 2, rel_heights =  c(1,1))

p.dotplot.proteins.intra &lt;- plot_grid(p.CD27IgD.dotplot.PROT.intra.sign,p.CD27IgD.dotplot.PROTsign.intra.neg,
          labels = c(panellabels[8:9]), label_size = 10, ncol = 2, rel_heights =  c(1,1))




p.supplement.percentages.dotplots &lt;- plot_grid(p.percentages,
                                               p.dotplot.genes,
                                               p.dotplot.proteins.surface,
                                               p.dotplot.proteins.intra,
                                               labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1.5,1.7, 1.25,1.15))

#ggsave(p.supplement.percentages.dotplots, filename = &quot;output/paper_figures/Suppl_CC_CD27_dotplots.eps&quot;, width = 177, height = 220, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
#ggsave(p.supplement.percentages.dotplots, filename = &quot;output/paper_figures/Suppl_CC_CD27_dotplots.TIFF&quot;, width = 177, height = 220, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>p.supplement.percentages.dotplots</code></pre>
<p><img src="figure/cellstate.Rmd/Suppl_CC_CD27_dotplots-1.png" width="691.2" style="display: block; margin: auto;" />
<strong>Supplementary figure</strong> Additional information
supplementing figure 1. <em>A.</em> fixed data percentage cycling.
<em>B-C.</em> Percentages CD27+IgD- cells in datasets. <em>D-J</em>
Differentially expressed genes and proteins.</p>
<div id="umap-on-shared-proteins" class="section level5">
<h5>UMAP on shared proteins</h5>
<pre class="r"><code>library(umap)

common.prot &lt;- left_join(live.DT.PROT.common,fix.DT.PROT.common)

umap.common.prot &lt;- umap(t(common.prot[,-&quot;gene&quot;]), random_state=42)


meta_inorder &lt;- data.frame( sample = colnames(common.prot[,-&quot;gene&quot;]))
meta_inorder &lt;- left_join(meta_inorder, meta.all)
umap.labels &lt;- meta_inorder$dataset

umap.common.prot.withmeta &lt;- cbind(data.frame(umap.common.prot$layout),umap.labels)

reviewer_umap_commonProt &lt;- ggplot(as.data.frame(umap.common.prot.withmeta), aes(x = X1, y = X2, fill = umap.labels ))+
  geom_point(size = 0.9, alpha = 0.8, shape = 21) +
  labs(title = &quot;UMAP on common proteins \nshows full overlap of the two datasets&quot;) +
scale_fill_manual(&quot;Dataset&quot;,values=c(&quot;lightgrey&quot;, &quot;black&quot;)) &amp;
    add.textsize &amp;  
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.25), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=12), axis.ticks = element_blank(),
          axis.text=element_text(size=12),
          plot.title = element_text(size=12, face = &quot;bold&quot;,hjust = 0)) 
#reviewer_umap_commonProt
# 
# ggsave(reviewer_umap_commonProt, filename = &quot;output/paper_figures/reviewer_umap_commonProt.png&quot;, width = 70, height = 80, units = &quot;mm&quot;,  dpi = 300)
# ggsave(reviewer_umap_commonProt, filename = &quot;output/paper_figures/reviewer_umap_commonProt.eps&quot;, width = 70, height = 80, units = &quot;mm&quot;,  dpi = 300)</code></pre>
</div>
</div>
<div id="supplement-mofa-properties-and-clustering"
class="section level3">
<h3>Supplement MOFA properties and clustering</h3>
<pre class="r"><code>p.mofa.suppl &lt;- plot_grid(
p.mofaoverview.input,p.variance.perfactor.all,reviewer_umap_commonProt,NULL,plot.rank.PROT,p.MOFA.factors.cluster,labels = c(panellabels[c(1:2)],panellabels[4],&quot;&quot;,panellabels[c(3,5)]), 
          label_size = 10,ncol = 3,
          rel_widths = c(0.5,2,1), rel_heights = c(1,1))

 ggsave(p.mofa.suppl, filename = &quot;output/paper_figures/Suppl_mofa.pdf&quot;, width = 483, height = 300, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
 ggsave(p.mofa.suppl, filename = &quot;output/paper_figures/Suppl_mofa.png&quot;, width = 483, height = 300, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>p.mofa.suppl</code></pre>
<p><img src="figure/cellstate.Rmd/fig-1.png" width="691.2" style="display: block; margin: auto;" />
<strong>Supplementary figure</strong> Additional information on MOFA
analysis</p>
</div>
<div id="supplement-seurat-protein-clustering" class="section level3">
<h3>Supplement Seurat protein clustering</h3>
<pre class="r"><code>fig.2.suppl.row1 &lt;- plot_grid(p.clusters.umap.live,  p.umap.Ig.prot.live,labels = panellabels[c(1,3)], label_size = 10, rel_widths = c(1,3))

fig.2.suppl.row2 &lt;- plot_grid(p.donor.umap.live, p.umap.Ig.RNA.live,labels = panellabels[2], label_size = 10,
          rel_widths = c(1,3))

p.umap.fix.all &lt;- plot_grid(p.clusters.umap.fix, 
          p.umap.Ig.prot.fix, labels = panellabels[c(4,5)], 
          label_size = 10,
          rel_widths = c(1,3))


fig.2.suppl.full &lt;- plot_grid(fig.2.suppl.row1, fig.2.suppl.row2,p.umap.fix.all, labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1,1,1))

#ggsave(fig.2.suppl.full, filename = &quot;output/paper_figures/Suppl_SeuratCluster.eps&quot;, width = 177, height = 180, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
#ggsave(fig.2.suppl.full, filename = &quot;output/paper_figures/Suppl_SeuratCluster.TIFF&quot;, width = 177, height = 180, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>fig.2.suppl.full</code></pre>
<p><img src="figure/cellstate.Rmd/fig.2.suppl.full-1.png" width="691.2" style="display: block; margin: auto;" /></p>
<p><strong>Supplementary figure</strong> Ig-class protein based
visualization</p>
<pre class="r"><code>## Run PCA without Ig
seu.fix &lt;- RunPCA(seu.fix, reduction.name = &#39;pcanoIG&#39;, features = rownames(seu.fix)[!(rownames(seu.fix) %in% grep(rownames(seu.fix),pattern = &quot;^Ig&quot;,value = TRUE,invert = FALSE))])

seu.fix &lt;- RunUMAP(seu.fix,dims = 1:4, reduction = &quot;pcanoIG&quot;, reduction.name = &quot;noIGUMAP&quot; )
seu.fix &lt;- FindNeighbors(seu.fix, reduction = &quot;pcanoIG&quot;,assay = &quot;PROT&quot;, dims = 1:4, graph.name = &quot;pcanoIG_nn&quot;)
seu.fix &lt;- FindClusters(seu.fix, graph.name = &quot;pcanoIG_nn&quot;,  algorithm = 3,resolution = 0.3, verbose = FALSE)
seu.fix[[&quot;clusters_pcanoIG&quot;]] &lt;- Idents(object = seu.fix)


p.umap.Ig.prot.fix2 &lt;- FeaturePlot(seu.fix, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;),slot = &quot;scale.data&quot;, reduction = &#39;noIGUMAP&#39;, max.cutoff = 2, 
                  cols = c(&quot;lightgrey&quot;,&quot;deeppink3&quot;), ncol = 3, pt.size = 0.5)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Scaled \ncounts&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.85), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))

p.clusters.umap.fix2 &lt;- DimPlot(seu.fix, reduction = &#39;noIGUMAP&#39;,group.by = &quot;clusters_pcaIG_named&quot;, label = TRUE, repel = TRUE, label.size = 2.5, cols = c(&quot;#009E73&quot;, &quot;#D55E00&quot;,&quot;#F0E442&quot;), pt.size = 0.5,
)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Ig-based \nclusters&quot;, title = &quot;Fixed cells PCA &amp; UMAP \nwithout Ig-proteins&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))



# seu.fix &lt;- RenameIdents(object = seu.fix, `1` = &quot;IgM&quot;, `2` = &quot;IgG&quot;, `3` = &quot;IgA&quot;)
# seu.fix[[&quot;clusters_pcaIG_named&quot;]] &lt;- Idents(object = seu.fix)

## ---- live cells

## Run PCA on Ig proteins and on Ig genes
seu.live &lt;- RunPCA(seu.live, assay = &quot;PROT&quot;,reduction.name = &#39;pcanoIGPROT&#39;, features = rownames(seu.live@assays$PROT)[!(rownames(seu.live@assays$PROT) %in% grep(rownames(seu.live@assays$PROT),pattern = &quot;^Ig&quot;,value = TRUE,invert = FALSE))])


seu.live &lt;- RunPCA(seu.live, assay = &quot;SCT&quot;,reduction.name = &#39;pcanoIGRNA&#39;, features = rownames(seu.live@assays$SCT)[!(rownames(seu.live@assays$SCT) %in% grep(rownames(seu.live@assays$SCT),pattern = &quot;^IG&quot;,value = TRUE,invert = FALSE))])


seu.live &lt;- RunUMAP(seu.live,dims = 1:10, reduction = &quot;pcanoIGPROT&quot;, reduction.name = &quot;pcanoIGPROT_UMAP&quot; )

p.clusters.umap.live_noIg_PROT &lt;- DimPlot(seu.live, reduction = &#39;pcanoIGPROT_UMAP&#39;,group.by = &quot;clusters_pcaIG_named&quot;, label = TRUE, repel = TRUE, label.size = 2.5,pt.size =  0.5, cols = c(&quot;#009E73&quot;, &quot;#D55E00&quot;,&quot;#F0E442&quot;,&quot;grey&quot;)
)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Clusters&quot;, title = &quot;Live cells UMAP based on \nprotein modality PCA \nwithout Ig-proteins&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(1, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))

p.umap.Ig.live_noIg_PROT &lt;- FeaturePlot(seu.live, features = c(&quot;IgM&quot;,&quot;IgA&quot;,&quot;IgG&quot;),slot = &quot;scale.data&quot;, reduction = &#39;pcanoIGPROT_UMAP&#39;, max.cutoff = 2, 
                  cols = c(&quot;lightgrey&quot;,&quot;deeppink3&quot;), ncol = 3, pt.size = 0.5)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))

seu.live &lt;- RunUMAP(seu.live,dims = 1:20, reduction = &quot;pcanoIGRNA&quot;, reduction.name = &quot;pcanoIGRNA_UMAP&quot; )



p.clusters.umap.live_noIg_RNA &lt;-DimPlot(seu.live, reduction = &#39;pcanoIGRNA_UMAP&#39;,group.by = &quot;clusters_pcaIG_named&quot;, label = TRUE, repel = TRUE, label.size = 2.5,pt.size =  0.5, cols = c(&quot;#009E73&quot;, &quot;#D55E00&quot;,&quot;#F0E442&quot;,&quot;grey&quot;)
)   &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;Clusters&quot;, title = &quot;Live cells UMAP based on \nRNA modality PCA \nwithout Ig-genes&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = &quot;none&quot;, legend.key.size = unit(1, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=6, face = &quot;bold&quot;))

p.umap.Ig.live_noIg_RNA &lt;- FeaturePlot(seu.live, features = c(&quot;IGHM&quot;,&quot;IGHA1&quot;,&quot;IGHG1&quot;),slot = &quot;scale.data&quot;, reduction = &#39;pcanoIGRNA_UMAP&#39;, max.cutoff = 3, 
                  cols = c(&quot;lightgrey&quot;,&quot;dodgerblue3&quot;), ncol = 3, pt.size =  0.5)  &amp;
  labs(x = &quot;UMAP 1&quot;, y = &quot;UMAP 2&quot;, color = &quot;&quot;) &amp;
  theme_half_open()&amp; 
  theme(legend.position = c(0.85,0.95), legend.key.size = unit(2, &#39;mm&#39;)) &amp;
  theme(axis.text.x = element_blank(),
          axis.text.y = element_blank(),
          text = element_text(size=7), axis.ticks = element_blank(),
          axis.text=element_text(size=7),
          plot.title = element_text(size=7, face = &quot;bold&quot;,hjust = 0.5))</code></pre>
<pre class="r"><code>fig.2.suppl.row1_2 &lt;- plot_grid(p.clusters.umap.fix2,  p.umap.Ig.prot.fix2,labels = panellabels[c(1,3)], label_size = 10, rel_widths = c(1,3))

fig.2.suppl.row2_2 &lt;- plot_grid(p.clusters.umap.live_noIg_RNA, p.umap.Ig.live_noIg_RNA,labels = panellabels[2], label_size = 10,
          rel_widths = c(1,3))

p.umap.fix.all_2 &lt;- plot_grid(p.clusters.umap.live_noIg_PROT, 
          p.umap.Ig.live_noIg_PROT, labels = panellabels[c(4,5)], 
          label_size = 10,
          rel_widths = c(1,3))


fig.2.suppl.full_2 &lt;- plot_grid(fig.2.suppl.row1_2, fig.2.suppl.row2_2,p.umap.fix.all_2, labels = &quot;&quot;, label_size = 10, ncol = 1, rel_heights = c(1,1,1))

 ggsave(fig.2.suppl.full_2, filename = &quot;output/paper_figures/Suppl_SeuratCluster_noIg.pdf&quot;, width = 177, height = 180, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
#  # ggsave(fig.2.suppl.full_2, filename = &quot;output/paper_figures/Suppl_SeuratCluster_noIg.TIFF&quot;, width = 177, height = 180, units = &quot;mm&quot;,  dpi = 300)
ggsave(fig.2.suppl.full_2, filename = &quot;output/paper_figures/Suppl_SeuratCluster_noIg.png&quot;, width = 177, height = 180, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>fig.2.suppl.full_2</code></pre>
<p><img src="figure/cellstate.Rmd/fig.2.suppl.full_noIg-1.png" width="691.2" style="display: block; margin: auto;" /></p>
<p><strong>Supplementary figure</strong> UMAP based on proteins/genes
without Ig’s.</p>
</div>
<div id="supplement-ig-class-and-elisa" class="section level3">
<h3>Supplement Ig-class % and ELISA</h3>
<p>Import ELISA results</p>
<pre class="r"><code>#pzfx_tables(&quot;data/supplementary/Figure 3/Fig3ELISA/GM6642 Ig ELISA.pzfx&quot;)

df.ELISA.IgA &lt;- read_pzfx(&quot;data/supplementary/Figure 3/Fig3ELISA/GM6642 Ig ELISA.pzfx&quot;, table = &quot;IgA production in time&quot;)

df.ELISA.IgA &lt;- df.ELISA.IgA %&gt;% gather(sampleID,signal,2:ncol(df.ELISA.IgA))%&gt;%
  mutate(sample = ROWTITLE,Cytokine = &quot;IgA&quot;)

df.ELISA.IgG &lt;- read_pzfx(&quot;data/supplementary/Figure 3/Fig3ELISA/GM6642 Ig ELISA.pzfx&quot;, table = &quot;IgG production in time&quot;)

df.ELISA.IgG &lt;- df.ELISA.IgG %&gt;% gather(sampleID,signal,2:ncol(df.ELISA.IgG))%&gt;%
  mutate(sample = ROWTITLE,Cytokine = &quot;IgG&quot;)

df.ELISA.IgM &lt;- read_pzfx(&quot;data/supplementary/Figure 3/Fig3ELISA/GM6642 Ig ELISA.pzfx&quot;, table = &quot;IgM production in time&quot;)

df.ELISA.IgM &lt;- df.ELISA.IgM %&gt;% gather(sampleID,signal,2:ncol(df.ELISA.IgM))%&gt;%
  mutate(sample = ROWTITLE,Cytokine = &quot;IgM&quot;)


df.ELISA.all &lt;- rbind(df.ELISA.IgM, df.ELISA.IgA) %&gt;%
  rbind(.,df.ELISA.IgG)%&gt;%
  separate(sampleID,c(&quot;sampleID&quot;, &quot;rep&quot;), sep = &quot;_&quot;) 

df.ELISA.all$sample &lt;- factor(df.ELISA.all$sample, levels = c(&quot;Day 0&quot; , &quot;Day 1&quot;,  &quot;Day 4&quot;,  &quot;Day 7&quot;,  &quot;Day 11&quot;))

summarySE &lt;- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA&#39;s: if na.rm==T, don&#39;t count them
    length2 &lt;- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group&#39;s data frame, return a vector with
    # N, mean, and sd
    datac &lt;- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the &quot;mean&quot; column    
    datac &lt;- rename(datac, c(&quot;mean&quot; = measurevar))

    datac$se &lt;- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult &lt;- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci &lt;- datac$se * ciMult

    return(datac)
}

tgc &lt;- summarySE(df.ELISA.all, measurevar=&quot;signal&quot;, groupvars=c(&quot;sample&quot;, &quot;Cytokine&quot;,&quot;sampleID&quot;))


p.ELISA &lt;- ggplot(tgc, aes(x=sample, y=signal, colour=sampleID, group=sampleID)) + geom_point( size=2, shape=21, fill=&quot;white&quot;) + # 21 is filled circle
    geom_errorbar(aes(ymin=signal-se, ymax=signal+se), colour=&quot;black&quot;, width=.1) +
    geom_line() +
    
    xlab(&quot;&quot;) +
    ylab(&quot;Concentration (ng/ml)&quot;) +
    scale_colour_hue(name=&quot;Donor&quot;)+
    theme_few()+
  facet_wrap(.~Cytokine, scales = &quot;free_y&quot;)+
  add.textsize+ theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<pre class="r"><code>Suppl.IgG.validation &lt;- plot_grid(p.percentages.Igclasses.MOFAclusters,p.ELISA, labels = panellabels[1:2], label_size = 10, ncol = 2, rel_widths = c(1,2))

#ggsave(Suppl.IgG.validation, filename = &quot;output/paper_figures/Suppl.IgG.validation.eps&quot;, width = 177, height = 60, units = &quot;mm&quot;,  dpi = 300, useDingbats = FALSE)
#ggsave(Suppl.IgG.validation, filename = &quot;output/paper_figures/Suppl.IgG.validation.TIFF&quot;, width = 177, height = 60, units = &quot;mm&quot;,  dpi = 300)</code></pre>
<pre class="r"><code>Suppl.IgG.validation</code></pre>
<p><img src="figure/cellstate.Rmd/Suppl.IgG.validation-1.png" width="691.2" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-dataset" class="section level2">
<h2>Save dataset</h2>
<p>Save seurat objects with selected non-cycling &amp; CD27+IgD-
cells.</p>
<pre class="r"><code>#saveRDS(seu.fix, &quot;output/seu.fix_norm_plasmacells.rds&quot;)
#saveRDS(seu.live, &quot;output/seu.live_norm_plasmacells.rds&quot;)

#saveRDS(mofa, file= &quot;output/MOFA_analysis_Donorgroup_clustered.rds&quot;)</code></pre>
<pre class="r"><code>seu.live.filtered.RNA &lt;- seu.live
DefaultAssay(seu.live.filtered.RNA) &lt;- &quot;RNA&quot;
seu.live.filtered.RNA[[&quot;SCT&quot;]] &lt;- NULL

seu.live.filtered.RNA[[&quot;PROT&quot;]] &lt;- NULL

#saveRDS(seu.live.filtered.RNA, &quot;output/seu.live_norm_plasmacells_RNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   
[3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] plyr_1.8.6             umap_0.2.9.0           MOFA2_1.1.17          
 [4] pzfx_0.3.0             ggupset_0.3.0          RColorBrewer_1.1-2    
 [7] clusterProfiler_3.18.1 enrichplot_1.10.2      UCell_1.0.0           
[10] data.table_1.14.2      scales_1.1.1           cowplot_1.1.1         
[13] ggthemes_4.2.4         kableExtra_1.3.4       knitr_1.36            
[16] org.Hs.eg.db_3.12.0    AnnotationDbi_1.52.0   IRanges_2.24.1        
[19] S4Vectors_0.28.1       Biobase_2.50.0         BiocGenerics_0.36.1   
[22] forcats_0.5.1          stringr_1.4.0          dplyr_1.0.7           
[25] purrr_0.3.4            readr_2.1.0            tidyr_1.1.4           
[28] tibble_3.1.5           ggplot2_3.3.5          tidyverse_1.3.1       
[31] Matrix_1.3-4           SeuratObject_4.0.2     Seurat_4.0.2          
[34] workflowr_1.6.2       

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3        scattermore_0.7       ragg_1.2.0           
  [4] bit64_4.0.5           irlba_2.3.3           DelayedArray_0.16.3  
  [7] rpart_4.1-15          generics_0.1.1        RSQLite_2.2.8        
 [10] shadowtext_0.0.9      RANN_2.6.1            future_1.23.0        
 [13] bit_4.0.4             tzdb_0.2.0            spatstat.data_2.1-0  
 [16] webshot_0.5.2         xml2_1.3.2            lubridate_1.8.0      
 [19] httpuv_1.6.3          assertthat_0.2.1      viridis_0.6.2        
 [22] xfun_0.26             hms_1.1.1             jquerylib_0.1.4      
 [25] evaluate_0.14         promises_1.2.0.1      fansi_0.5.0          
 [28] dbplyr_2.1.1          readxl_1.3.1          igraph_1.2.6         
 [31] DBI_1.1.1             htmlwidgets_1.5.4     spatstat.geom_2.2-2  
 [34] ellipsis_0.3.2        corrplot_0.92         RSpectra_0.16-0      
 [37] backports_1.3.0       deldir_1.0-2          MatrixGenerics_1.2.1 
 [40] vctrs_0.3.8           ROCR_1.0-11           abind_1.4-5          
 [43] cachem_1.0.6          withr_2.5.0           ggforce_0.3.3        
 [46] sctransform_0.3.2     goftest_1.2-2         svglite_2.0.0        
 [49] cluster_2.1.0         DOSE_3.16.0           lazyeval_0.2.2       
 [52] crayon_1.4.2          basilisk.utils_1.2.2  pkgconfig_2.0.3      
 [55] labeling_0.4.2        tweenr_1.0.2          nlme_3.1-149         
 [58] rlang_0.4.11          globals_0.14.0        lifecycle_1.0.1      
 [61] miniUI_0.1.1.1        downloader_0.4        filelock_1.0.2       
 [64] modelr_0.1.8          cellranger_1.1.0      rprojroot_2.0.2      
 [67] polyclip_1.10-0       matrixStats_0.61.0    lmtest_0.9-38        
 [70] Rhdf5lib_1.12.1       zoo_1.8-9             reprex_2.0.1         
 [73] whisker_0.4           ggridges_0.5.3        pheatmap_1.0.12      
 [76] png_0.1-7             viridisLite_0.4.0     KernSmooth_2.23-17   
 [79] rhdf5filters_1.2.1    blob_1.2.2            qvalue_2.22.0        
 [82] parallelly_1.29.0     memoise_2.0.1         magrittr_2.0.1       
 [85] ica_1.0-2             compiler_4.0.3        scatterpie_0.1.7     
 [88] fitdistrplus_1.1-6    cli_3.6.0             listenv_0.8.0        
 [91] patchwork_1.1.1       pbapply_1.5-0         MASS_7.3-53          
 [94] mgcv_1.8-33           tidyselect_1.1.1      stringi_1.7.5        
 [97] textshaping_0.3.6     highr_0.9             yaml_2.2.1           
[100] GOSemSim_2.16.1       askpass_1.1           ggrepel_0.9.1        
[103] grid_4.0.3            sass_0.4.0            fastmatch_1.1-3      
[106] tools_4.0.3           future.apply_1.8.1    rstudioapi_0.13      
[109] git2r_0.28.0          gridExtra_2.3         farver_2.1.0         
[112] Rtsne_0.15            ggraph_2.0.5          digest_0.6.28        
[115] rvcheck_0.2.1         BiocManager_1.30.16   shiny_1.7.1          
[118] Rcpp_1.0.7            broom_0.7.10          later_1.3.0          
[121] RcppAnnoy_0.0.19      httr_1.4.2            colorspace_2.0-2     
[124] rvest_1.0.2           fs_1.5.0              tensor_1.5           
[127] reticulate_1.22       splines_4.0.3         uwot_0.1.10          
[130] yulab.utils_0.0.4     spatstat.utils_2.2-0  graphlayouts_0.7.2   
[133] basilisk_1.2.1        plotly_4.10.0         systemfonts_1.0.3    
[136] xtable_1.8-4          jsonlite_1.7.2        tidygraph_1.2.0      
[139] ggfun_0.0.4           R6_2.5.1              pillar_1.6.4         
[142] htmltools_0.5.2       mime_0.12             glue_1.4.2           
[145] fastmap_1.1.0         BiocParallel_1.24.1   codetools_0.2-16     
[148] fgsea_1.16.0          utf8_1.2.2            lattice_0.20-41      
[151] bslib_0.3.1           spatstat.sparse_2.0-0 leiden_0.3.9         
[154] openssl_1.4.5         GO.db_3.12.1          survival_3.2-7       
[157] limma_3.46.0          rmarkdown_2.11        munsell_0.5.0        
[160] DO.db_2.9             rhdf5_2.34.0          HDF5Array_1.18.1     
[163] haven_2.4.3           reshape2_1.4.4        gtable_0.3.0         
[166] spatstat.core_2.3-0  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
